### ISSUE TITLE
TypeError when calling BlockVecSubVectorWrapper

### ISSUE THREAD

**Issue by EngineerEricXie** [multiphenics/multiphenicsx#41]
I am running Tutorial 06, case 2a: advection diffusion reaction problem with distributed control on my Windows PC in wsl. The code runs smoothly until it reaches the 22 code block, where multiphenicsx.fem.petsc.BlockVecSubVectorWrapper() were called. The following are the error message shown in the output. I have checked the petsc.py file and make sure that the first and second argument are "list", which should satisfy the collections.abc.Sequence input datatype. I am wondering what the error could possibly be?

I ran the same tutorial on google Colab and there is no error. And the version of each package are listed below.
multiphenicsx = 0.3.dev1
fenics-dolfinx = 0.9.0

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[22], [line 2](vscode-notebook-cell:?execution_count=22&line=2)
      [1](vscode-notebook-cell:?execution_count=22&line=1) # Split the block solution in components
----> [2](vscode-notebook-cell:?execution_count=22&line=2) with multiphenicsx.fem.petsc.BlockVecSubVectorWrapper(yup, [Y.dofmap, U.dofmap, Q.dofmap]) as yup_wrapper:
      [3](vscode-notebook-cell:?execution_count=22&line=3)     for yup_wrapper_local, component in zip(yup_wrapper, (y, u, p)):
      [4](vscode-notebook-cell:?execution_count=22&line=4)         with component.x.petsc_vec.localForm() as component_local:

File ~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:483, in BlockVecSubVectorWrapperBase.<locals>.BlockVecSubVectorWrapperBase_Class.__init__(self, b, dofmaps, restriction, ghosted)
    [481](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:481) if restriction is None:
    [482](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:482)     index_maps = [(dofmap.index_map, dofmap.index_map_bs) for dofmap in dofmaps]
--> [483](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:483)     index_sets = mcpp.la.petsc.create_index_sets(
    [484](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:484)         index_maps, [1] * len(index_maps), ghosted=ghosted,
    [485](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:485)         ghost_block_layout=mcpp.la.petsc.GhostBlockLayout.trailing)
    [486](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:486)     self._unrestricted_index_sets = index_sets
    [487](https://vscode-remote+wsl-002bubuntu-002d22-002e04.vscode-resource.vscode-cdn.net/home/tsungyeh/multiphenicsx/tutorials/06_optimal_control/~/anaconda3/envs/fenics/lib/python3.10/site-packages/multiphenicsx/fem/petsc.py:487)     self._restricted_index_sets = None

TypeError: create_index_sets(): incompatible function arguments. The following argument types are supported:
    1. create_index_sets(maps: collections.abc.Sequence[tuple[dolfinx::common::IndexMap, int]], is_bs: collections.abc.Sequence[int], ghosted: bool = True, ghost_block_layout: multiphenicsx.cpp.multiphenicsx_cpp.la.petsc.GhostBlockLayout = GhostBlockLayout.intertwined) -> list[object]

Invoked with types: list, list, kwargs = { ghosted: bool, ghost_block_layout: multiphenicsx.cpp.multiphenicsx_cpp.la.petsc.GhostBlockLayout }

**Comment by francesco-ballarin** [comment]
For future post, please format corrrectly your error (using the "Code" button), otherwise it's very difficult to read.

That sort of error typically means that you have a mismatch between the nanobind version that was used while building dolfinx, and the nanobind version you are using right now to build multiphenicsx. How did you install nanobind? How did you install nanobind (if you installed it yourself)?

**Comment by EngineerEricXie** [comment]
@francesco-ballarin Thank you for your reply, and I apologize for the messy error format earlier.

The issue was indeed caused by a version mismatch with `nanobind`. Initially, I installed it using `pip install nanobind` without specifying a version. When I later ran `pip install nanobind==2.1.0`, I discovered that my Python version was too old. After upgrading the Python version in my environment and reinstalling `nanobind`, the issue was resolved. Perhaps it would be helpful to add a note about compatible Python versions on the multiphenicsx installation webpage.

Additionally, when running in WSL, I encountered errors related to missing libraries when importing gmsh. The error messages were:

```
- OSError: libGLU.so.1: cannot open shared object file: No such file or directory
- OSError: libXcursor.so.1: cannot open shared object file: No such file or directory
```

These errors were resolved by running the following commands:

```
- sudo apt-get install libglu1-mesa  
- sudo apt-get install libxcursor1 libxcursor-dev
```
